File: backend/src/main/java/com/example/demo/entity/ProductColor.java
package com.example.demo.entity;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import java.util.Date;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "product_color")
public class ProductColor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "product_id")
    @JsonIgnore
    private Product product;

    @ManyToOne
    @JoinColumn(name = "color_id")
    private Color color;

    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    private Date lastModifiedAt;

    private String createdBy;
}

File: frontend/types/product-color.ts
import { Color } from "./color";
import { ProductForm } from "./product";

export type ProductColor = {
  id?: number;
  createdAt?: Date;
  lastModifiedAt?: Date;
  createdBy?: string;
  color: Color;
  product: ProductForm;
};

export type GetColorImgFacade = {
  colorId: number;
  productId: number;
};
